File: src/pocketcloud/cloud/network/packet/impl/normal/ProxyUnregisterServerPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\normal;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\data\PacketData;

class ProxyUnregisterServerPacket extends CloudPacket {

    public function __construct(private string $serverName = "") {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->write($this->serverName);
    }

    public function decodePayload(PacketData $packetData): void {
        $this->serverName = $packetData->readString();
    }

    public function getServerName(): string {
        return $this->serverName;
    }

    public function handle(ServerClient $client): void {}

    public static function create(string $server): self {
        return new self($server);
    }
}

File: cloud-network/src/main/java/de/pocketcloud/network/packet/impl/ProxyUnregisterServerPacket.java
package de.pocketcloud.network.packet.impl;

import de.pocketcloud.api.model.server.ICloudServer;
import de.pocketcloud.api.network.packet.ClientboundPacket;
import de.pocketcloud.api.network.packet.data.IPacketData;
import de.pocketcloud.network.packet.CloudPacket;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public final class ProxyUnregisterServerPacket extends CloudPacket implements ClientboundPacket {

    private String serverName;

    public ProxyUnregisterServerPacket(String serverName) {
        this.serverName = serverName != null ? serverName : "";
    }

    @Override
    public void encodePayload(IPacketData packetData) {
        packetData.writeAll(serverName);
    }

    @Override
    public void decodePayload(IPacketData packetData) {}

    public static ProxyUnregisterServerPacket create(String serverName) {
        return new ProxyUnregisterServerPacket(serverName);
    }

    public static ProxyUnregisterServerPacket create(ICloudServer server) {
        return new ProxyUnregisterServerPacket(server.name());
    }
}
